src/qibolab/pulses.py

```diff
@@ -178,6 +178,10 @@ def __init__(self):
         self.name = "Rectangular"
         self.pulse: Pulse = None
 
+    def __eq__(self, item) -> bool:
+        """Overloads == operator"""
+        return type(item) is Rectangular
```

Comment:
Another more common way to check types is
```suggestion
        return isinstance(item, Rectangular)
```
or in this case
```suggestion
        return isinstance(item, self.__class__)
```
but I am not sure which is better. 

Here is a long discussion of `type` vs `isinstance`: https://stackoverflow.com/questions/1549801/what-are-the-differences-between-type-and-isinstance

Reply:
From the link you sent: "isinstance caters for inheritance (an instance of a derived class is an instance of a base class, too), while checking for equality of type does not".
So I think it's better to check type equality, to avoid any weird inheritance problem.

But I think that using `self.__class__` is a good idea... maybe I can check  that equality in the abstract PulseShape object but I'm not sure how inheritance will behave